internal/functions/serve/serve.go

```diff
@@ -174,6 +174,11 @@ func Run(ctx context.Context, slug string, envFilePath string, noVerifyJWT *bool
 			} else {
 				return fmt.Errorf("failed to check index.ts for function %s: %w", slug, err)
 			}
+
+			if *unstable {
+				denoRunCmd = append(denoRunCmd, "--unstable")
```

Comment:
Have you tried deploying the puppeteer example with `supabase functions deploy`? If that works, we can always add the `--unstable` flag to deno run by default.

Ideally the same code should work in both local serving and production deployment.

Reply:
For probably unrelated reasons, the puppeteer example doesn't work locally all for me (even on official cli), there's some issue with readable streams that doesn't occur when running the same with `deno run` -
```
Error: channel closed
    at async UserWorker.fetch (ext:sb_user_workers/user_workers.js:54:21)
    at async Server.<anonymous> (file:///home/deno/main/index.ts:104:16)
    at async Server.#respond (https://deno.land/std@0.182.0/http/server.ts:220:24)
worker thread panicked Uncaught SyntaxError: The requested module '/v124/jszip@3.5.0/esnext/lib/readable-stream-browser.js' does not provide an export named 'default'
    at https://esm.sh/v124/jszip@3.5.0/esnext/jszip.mjs:2:291
worker "/home/deno/functions/puppeteer" returned an error: Uncaught SyntaxError: The requested module '/v124/jszip@3.5.0/esnext/lib/readable-stream-browser.js' does not provide an export named 'default'
    at https://esm.sh/v124/jszip@3.5.0/esnext/jszip.mjs:2:291
```

... but since it works great with `deno deploy` - https://gqlatxhpnxwuzspuqqvh.functions.supabase.co/puppeteer, and `deno bundle` and `deno run` both support `--unstable`, maybe it could be the default behavior?

Reply:
I'm afraid Deno KV is not currently supported by supabase functions because we are switching to [edge-runtime](https://github.com/supabase/edge-runtime) for both serving locally and prod deployment. The `--unstable` flag added in this PR targets the legacy deno relay server which we have deprecated.

Please consider opening a [github issue](https://github.com/supabase/edge-runtime/issues/new/choose) with edge runtime repo to help our functions team prioritise Deno KV support.